test/framework/src/main/java/org/elasticsearch/index/shard/IndexShardTestCase.java

```diff
@@ -1010,6 +1010,7 @@ protected ShardGeneration snapshotShard(final IndexShard shard, final Snapshot s
                     snapshotStatus,
                     Version.CURRENT,
                     Collections.emptyMap(),
+                    threadPool.absoluteTimeInMillis(),
```

Comment:
Also randomize here instead? I would think we do not rely on the specific time here.

Reply:
I don't think the test relies on the time. I've changed it to random, here and also all the following ones.